app/jobs/send_assessments_job.rb

```diff
@@ -5,15 +5,70 @@ class SendAssessmentsJob < ApplicationJob
   queue_as :mailers
 
   def perform(*_args)
-    eligible = Patient.reminder_eligible.count
-    sent = []
+    patient_ids = Patient.reminder_eligible.pluck(:id)
+    eligible = patient_ids.size
+
+    # Check what the max thread pool is for the DB since that is the limiting factor
+    # on how many threads can be used in the job. Subtracting 2 from the pool
+    # size to be on the safer side of exhausting the connection pool.
+    total_threads = [1, ActiveRecord::Base.connection_pool.stat[:size] - 2].max
```

Comment:
I like this, though I think the connection pool is ultimately controlled by an environment variable. Thoughts on doing the same for this? FWIW, as a rule of thumb I typically try to keep this sort of thing to 2^n.

Reply:
I thought the environment variable could override the default for the connection pool. So if the environment variable is not set then it will default to a pool of 10 I believe (see database.yml). I'm getting pool size from the `ActiveRecord::Base.connection_pool.stat[:size]` call so that I can always get the current pool size regardless of if the env variable is set or not